packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -224,11 +246,89 @@ class AuroraClusterEngine extends MySqlClusterEngineBase {
   }
 }
 
+/**
+ * The versions for the Aurora MySQL cluster engine
+ * (those returned by {@link DatabaseClusterEngine.auroraMysql}).
+ */
+export class AuroraMysqlEngineVersion {
+  /** Version "5.7.12". */
+  public static readonly VER_5_7_12 = AuroraMysqlEngineVersion.builtIn_5_7('12', false);
+  /** Version "5.7.mysql_aurora.2.03.2". */
+  public static readonly VER_2_03_2 = AuroraMysqlEngineVersion.builtIn_5_7('2.03.2');
```

Comment:
How'd you decide on the cut-off? I'm a bit concerned signing up for tracking each of the Aurora releases -- and maintaining each version here for $TIME_PERIOD_X until it can be deprecated and removed -- might be more maintenance than is advisable. There were 16 new versions of 2.x and 9 new versions of 1.x in 2019, so on average we may be updating this list twice a month.

Reply:
I don't think it's a big deal. We can just deprecate fields as we go along, and thanks to the `of` methods, we're not blocking customers from accessing new versions as they happen.

We do something [similar in CodeBuild](https://github.com/aws/aws-cdk/blob/928f37bddb41b747e7ab82dd14d0c0c8eb0530f2/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L1266-L1338), it hasn't been a big burden.